### ISSUE TITLE
NetworkInterface.GetAllNetworkInterfaces() throw exceptions in some docker environments

### ISSUE THREAD

**Issue by toddschilling** [MichaCo/DnsClient.NET#3]
There's a know defect where GetAllNetworkInterfaces() throws an error when it doesn't have privileged access to the filesystem in docker.  I'd recommend wrapping that call with its own exception handler and just parsing /etc/resolv.conf for nameservers as a final approach.

As a work around, I've done just that and then passed the resulting nameservers into your LookupClient for service discovery.

**Comment by MichaCo** [comment]
@toddschilling could you post the stacktrace? Because I actually have that implemented as a fallback, just never tested it in a docker container I think. 

here https://github.com/MichaCo/DnsClient.NET/blob/dev/src/DnsClient/NameServer.cs#L125

Just wondering maybe the `RuntimeInformation.IsOSPlatform` is not working properly...

Thanks
M

**Comment by toddschilling** [comment]
@MichaCo thanks for the quick response.  It appears as though the exception is falling through.
```
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6454 Unhandled Exception: System.UnauthorizedAccessException: Access to the path '/sys/class/net/lo/operstate' is denied. ---> System.IO.IOException: Permission denied
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    --- End of inner exception stack trace ---
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at System.IO.UnixFileStream.CheckFileCall(Int64 result, Boolean ignoreNotSupported)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at System.IO.UnixFileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at System.IO.UnixFileSystem.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at System.IO.File.InternalReadAllText(String path, Encoding encoding)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at System.IO.File.ReadAllText(String path)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at System.Net.NetworkInformation.LinuxNetworkInterface.GetOperationalStatus(String name)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at System.Net.NetworkInformation.LinuxNetworkInterface..ctor(String name)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at System.Net.NetworkInformation.LinuxNetworkInterface.GetOrCreate(Dictionary`2 interfaces, String name)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at System.Net.NetworkInformation.LinuxNetworkInterface.<>c__DisplayClass5_0.<GetLinuxNetworkInterfaces>b__2(String name, LinkLayerAddressInfo* llAddr)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at Interop.Sys.EnumerateInterfaceAddresses(IPv4AddressDiscoveredCallback ipv4Found, IPv6AddressDiscoveredCallback ipv6Found, LinkLayerAddressDiscoveredCallback linkLayerFound)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at System.Net.NetworkInformation.LinuxNetworkInterface.GetLinuxNetworkInterfaces()
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at DnsClient.NameServer.QueryNetworkInterfaces(Boolean skipIPv6SiteLocal)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at DnsClient.NameServer.ResolveNameServersInternal(Boolean skipIPv6SiteLocal)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at DnsClient.NameServer.ResolveNameServers(Boolean skipIPv6SiteLocal)
[build/my-service-discovery-2104d6ce-0-1491233850650146306] 15:38:26.6455    at mycode.init()
```

**Comment by MichaCo** [comment]
yeah I think I should just change the code and catch any exception and then fall back to read from file. 
I'm not checking for `UnauthorizedAccessException`...

There is not much to the catching anyways... 
Thanks for finding that ;)

**Comment by toddschilling** [comment]
no problem.  let me know when you get the new version on nuget.

**Comment by MichaCo** [comment]
@toddschilling should be fixed now in v1.0.6. Let me know how it goes ;)
